application/models/message_fixtures_test.go

```diff
@@ -180,3 +184,204 @@ func Fixtures_Message_FindByUUID(ms *ModelSuite) MessageFixtures {
+		{
+			Uuid:   domain.GetUuid(),
+			PostID: posts[0].ID,
+		},
+		{
+			Uuid:   domain.GetUuid(),
+			PostID: posts[1].ID,
+		},
+	}
+
+	for i := range threads {
+		createFixture(ms, &threads[i])
+	}
+
+	tNow := time.Now().Round(time.Duration(time.Second))
+	oldTime := tNow.Add(-time.Duration(time.Hour))
+	oldOldTime := oldTime.Add(-time.Duration(time.Hour))
+
+	// One thread per post with 2 users per thread
+	threadParticipants := []ThreadParticipant{
+		{
+			ThreadID:     threads[0].ID,
+			UserID:       posts[0].CreatedByID,
+			LastViewedAt: tNow, // This will get overridden and then reset again
+		},
+		{
+			ThreadID:     threads[0].ID,
+			UserID:       posts[0].ProviderID.Int,
+			LastViewedAt: oldTime,
+		},
+		{
+			ThreadID:     threads[1].ID,
+			UserID:       posts[1].CreatedByID,
+			LastViewedAt: oldTime,
+		},
+		{
+			ThreadID:     threads[1].ID,
+			UserID:       posts[1].ProviderID.Int,
+			LastViewedAt: tNow,
+		},
+	}
+
+	for i := range threadParticipants {
+		createFixture(ms, &threadParticipants[i])
+	}
+
+	// I can't seem to give them custom times
+	messages := Messages{
+		{
+			Uuid:      domain.GetUuid(),
+			ThreadID:  threads[0].ID,        // user 0's post
+			SentByID:  posts[0].CreatedByID, // user 0 (Eager)
+			Content:   "I can being chocolate if you bring PB",
+			CreatedAt: oldOldTime,
+		},
+		{
+			Uuid:      domain.GetUuid(),
+			ThreadID:  threads[0].ID,           // user 0's post
+			SentByID:  posts[0].ProviderID.Int, // user 1 (Lazy)
+			Content:   "Great",
+			CreatedAt: oldTime,
+		},
+		{
+			Uuid:      domain.GetUuid(),
+			ThreadID:  threads[0].ID,        // user 0's post
+			SentByID:  posts[0].CreatedByID, // user 0 (Eager)
+			Content:   "Can you get it here by next week?",
+			CreatedAt: tNow, // Lazy User doesn't see this one
+		},
+		{
+			Uuid:      domain.GetUuid(),
+			ThreadID:  threads[1].ID,        // user 1's post
+			SentByID:  posts[1].CreatedByID, // user 1 (Lazy)
+			Content:   "I can being PB if you bring chocolate",
+			CreatedAt: oldTime,
+		},
+		{
+			Uuid:      domain.GetUuid(),
+			ThreadID:  threads[1].ID,           // user 1's post
+			SentByID:  posts[1].ProviderID.Int, // user 0 (Eager)
+			Content:   "Did you see my other message?",
+			CreatedAt: tNow, // Lazy User doesn't see this one
+		},
+		{
+			Uuid:      domain.GetUuid(),
+			ThreadID:  threads[1].ID,           // user 1's post
+			SentByID:  posts[1].ProviderID.Int, // user 0 (Eager)
+			Content:   "Anyone Home?",
+			CreatedAt: tNow, // Lazy User doesn't see this one
+		},
+	}
+
+	for _, m := range messages {
+		if err := ms.DB.RawQuery(`INSERT INTO messages (content, created_at, sent_by_id, thread_id, updated_at, uuid)`+
+			`VALUES (?, ?, ?, ?, ?, ?)`,
+			m.Content, m.CreatedAt, m.SentByID, m.ThreadID, m.CreatedAt, m.Uuid).Exec(); err != nil {
+			t.Errorf("error loading messages ... %v", err)
+			t.FailNow()
+		}
+	}
+
+	// Because of Message.AfterCreate() we need to reset the LastViewedAt attributes
```

Comment:
Really? I'm surprised that `RawQuery` triggers `AfterCreate`.

Reply:
Oops, I should have taken out that code after I switched from using pop to RawQuery.